src/main/resources/workerScripts/fromMultiplePriorityQueues.lua

```diff
@@ -16,7 +18,7 @@ for q in queues.gmatch(queues, NEXT_QUEUE_REGEX) do
     local status, queueType = next(redis.call('TYPE', queueName))
     local payload
     if queueType == 'zset' then
-        local firstMsg = redis.call('ZRANGE', queueName, '0', '0')
+	    local firstMsg = redis.call('ZRANGEBYSCORE', queueName, '-inf', now, 'WITHSCORES', 'LIMIT' , '0' , '1')
```

Comment:
OK, sorry, I am not too familiar with the entire flow and in lua in general.
So just validating I got it right:
We use ZRANGEBYSCORE because we want to get a sorted set.
We wanna get the sorted set From queueName.
The -inf to now is the ranges we wanna fetch the data for: I assume these are the keys, right ?
WITHSCORE is added because you want the returned value to be returned with its score as well, right ?
LIMIT is simply to get the first element (0 for 0 offset and 1 to ge a set of 1 value, which is the first element).

did I get it right ?

If so: just one question: how and where do we define the scores ?